src/conversion/model-to-view-converters.js

```diff
@@ -478,6 +385,63 @@ export function remove() {
 	};
 }
 
+/**
+ * Function factory, crates marker to virtual selection converter.
+ * For more information see {@link module:engine/conversion/buildmodelconverter~ModelConverterBuilder#toVirtualSelection}.
+ *
+ * @param {module:engine/conversion/buildmodelconverter~VirtualSelectionDescriptor|Function} selectionDescriptor
+ * @return {Function} Marker to virtual selection converter.
+ */
+export function markerToVirtualSelection( selectionDescriptor ) {
+	return ( evt, data, consumable, conversionApi ) =>	{
+		const descriptor = typeof selectionDescriptor == 'function' ?
+			selectionDescriptor( data, consumable, conversionApi ) :
+			selectionDescriptor;
+
+		if ( !descriptor ) {
+			return;
+		}
+
+		const addMarker = evt.name.split( ':' )[ 0 ] == 'addMarker';
+		const modelItem = data.item;
+
+		// Return if model item is not present. This happens when whole marker conversion is fired before converting
+		// children of marker's range.
+		if ( !modelItem ) {
+			return;
+		}
+
+		if ( modelItem.is( 'textProxy' ) ) {
+			const viewElement = virtualSelectionDescriptorToAttribute( descriptor );
+			const converter = addMarker ?
+				wrapItem( viewElement, eventName => eventName ) :
+				unwrapItem( viewElement, eventName => eventName );
+
+			converter( evt, data, consumable, conversionApi );
```

Comment:
I got it on the second try :D So it's not that bad. The thing which confused me was the "...ToAttribute()" method because it doesn't return an attribute (key or value or something) but an element. The rest is fine (if you understand what wrapItem's second param does).